e2e/rescheduling/rescheduling_test.go

```diff
@@ -0,0 +1,513 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package rescheduling
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/nomad/e2e/e2eutil"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/jobspec"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/shoenig/test"
+	"github.com/shoenig/test/must"
+	"github.com/shoenig/test/wait"
+)
+
+const ns = "default"
+
+func reschedulingTestCleanup(t *testing.T, jobID string) func() {
+	return func() {
+		t.Helper()
+
+		if os.Getenv("NOMAD_TEST_SKIPCLEANUP") == "1" {
+			return
+		}
+
+		e2eutil.StopJob(jobID, "-purge", "-detach")
+		_, err := e2eutil.Command("nomad", "system", "gc")
+		test.NoError(t, err)
+	}
+}
```

Comment:
super nit: since this is in the `rescheduling` package, that could be excluded from this func name

and differently nit-picky for helper design, I'd write this like:

```suggestion
func cleanupJob(t *testing.T, jobID string) {
	if os.Getenv("NOMAD_TEST_SKIPCLEANUP") == "1" {
		return
	}
	t.Helper()
	t.Cleanup(func() {
		e2eutil.StopJob(jobID, "-purge", "-detach")
		_, err := e2eutil.Command("nomad", "system", "gc")
		test.NoError(t, err)
	})
}
```

then test cases could just call

```go
cleanupJob(t, jobID)
```

Reply:
Works for me, will do.

Reply:
it would return early from the `cleanupJob` func, prior to adding `t.Cleanup`, but the caller of `cleanupJob` would continue, eh?

Reply:
:facepalm:

Reply:
having it inside of `t.Cleanup` is just fine too though -- 6 of one, half a dozen of the other :yum:

Reply:
Fixed in 46a2c3e